query/excluder.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+//Excluder allows us to support different methods of excluding in the future
+type Excluder interface {
+	ShouldExclude(path string) bool
+}
+
+//RegexpExclude uses regular expressions to tell if a file/path should be excluded
+type RegexpExclude struct {
+	recursive  bool
+	exclusions []string
+	regex      string
+}
+
+//ShouldExclude will return a boolean denoting whether or not the path should be excluded based on the given slice of exclusions
+func (r *RegexpExclude) ShouldExclude(path string) bool {
+	var b bool
+	var ok error
+	if r.regex == "" {
```

Comment:
When no exclusion is provided, this is running every single time and nothing is being outputted (since everything matches `""`).

The solution here would be to track if the expression has been built or not, and if it's been built and still is empty, then this method should return false without even running `regexp.MatchString`.